WikipediaLibrary/TWLight
# If an authorization is to an EMAIL, CODES, or LINK authorization method partner, the Renew/Extend button should always be accessible

## Description
Changed the behavior of the `apply` and `has_auths` flags so that the Apply button is enabled for partners with certain authorization methods.

Fixed the resource tile partial view to show the `Renew`, `Extend` or `View application` buttons on individual access resources.

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
Before, it was impossible to ask for renewal if the editor already had authorization (even if It had already expired).

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
[T271088](https://phabricator.wikimedia.org/T271088)
[T270649](https://phabricator.wikimedia.org/T270649)

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
1. Create an authorization from a partner with  EMAIL, CODES, or LINK authorization method.
2. Create another authorization from a partner with  EMAIL, CODES, or LINK authorization method.
3. Create one pending application for one of the two partners.
4. Check that the `Apply` button for the partner with the pending application is disabled.
5. Check that the `Apply` button for the partner without the pending application is enabled.

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)
**Partner with Authorization Method EMAIL that has no open applications**
![Screen Shot 2021-03-12 at 21 36 39](https://user-images.githubusercontent.com/7854953/111017819-2e8ada80-837b-11eb-8675-b43490167f82.png)



**Partner with Authorization Method EMAIL that has an open application**
![Screen Shot 2021-03-12 at 21 36 55](https://user-images.githubusercontent.com/7854953/111017827-39456f80-837b-11eb-935b-85980965cd21.png)


**My Library view with the Extend and View Application buttons showing**
![Screen Shot 2021-03-15 at 19 55 07](https://user-images.githubusercontent.com/7854953/111244417-78b4cb80-85c8-11eb-97e1-aa550ab5cda4.png)


## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
